src/vs/workbench/contrib/search/browser/searchActionsTextQuickAccess.ts

```diff
@@ -29,6 +34,59 @@ registerAction2(class TextSearchQuickAccessAction extends Action2 {
 
 	override async run(accessor: ServicesAccessor, match: RenderableMatch | undefined): Promise<any> {
 		const quickInputService = accessor.get(IQuickInputService);
-		quickInputService.quickAccess.show(TEXT_SEARCH_QUICK_ACCESS_PREFIX);
+		const searchText = getSearchText(accessor);
+		const searchContents = searchText ? ' ' + searchText : '';
+		quickInputService.quickAccess.show(TEXT_SEARCH_QUICK_ACCESS_PREFIX + searchContents);
 	}
 });
+
+function getSearchText(accessor: ServicesAccessor): string | null {
+	const editorService = accessor.get(IEditorService);
+	const configurationService = accessor.get(IConfigurationService);
+
+	if (editorService.activeTextEditorControl === undefined) {
+		return null;
+	}
+	const activeEditor: IEditor = editorService.activeTextEditorControl;
+	if (!activeEditor) {
+		return null;
+	}
+	if (!activeEditor.hasTextFocus()) {
+		return null;
+	}
+
+	// only happen if it would also happen for the search view
+	const seedSearchStringFromSelection = configurationService.getValue<IEditorOptions>('editor').find!.seedSearchStringFromSelection;
+	if (!seedSearchStringFromSelection) {
+		return null;
+	}
+
+	const range = activeEditor.getSelection();
+	if (!range) {
```

Comment:
Same here, I prefer conditionals to be combined when they have the same behavior (unless it makes readability much worse).